compiler/rustc_lint/src/builtin.rs

```diff
@@ -548,17 +548,9 @@ impl<'tcx> LateLintPass<'tcx> for MissingDoc {
 
     fn check_item(&mut self, cx: &LateContext<'_>, it: &hir::Item<'_>) {
         match it.kind {
```

Comment:
All exported items should generally be checked, so keeping a whitelist like this doesn't seem right.

I'd remove this `match` entirely.
If this leads to unwanted lints to be reported for some `ItemKind`s then we should list those exceptions explicitly and add comments explaining why the lints are not wanted for these specific items.

Reply:
The only lints that throw errors are for `ItemKind::Impl` and `ItemKind::Use` , I would like to add comments for the exceptions but I don't understand what the reasoning is

Reply:
I guess the reasoning "we didn't report them historically, so we continue to not report them now for backward compatibility" would be enough.